<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /home/mnaveau/devel/workspace/src/catkin/robots/robot_properties/robot_properties_quadruped/xacro/quadruped_visual.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="quadruped" xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller" xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface" xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor" xmlns:xacro="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface">
  <!-- Inertia computation -->
  <link name="base_link">
    <!-- BASE LINK INERTIAL -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="2.0"/>
      <inertia ixx="0.00708333333333" ixy="0" ixz="0" iyy="0.0270833333333" iyz="0" izz="0.0333333333333"/>
    </inertial>
    <!-- BASE LINK VISUAL -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robot_properties_quadruped/meshes/obj/quadruped_body.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </visual>
    <!-- BASE LINK COLLISION -->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robot_properties_quadruped/meshes/obj/quadruped_body.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </collision>
  </link>
  <!-- END BASE LINK -->
  <!-- Include the legs -->
  <!-- See TODO: add the wiki link here -->
  <!-- These are absoulte distance value, the sign are decided below -->
  <!-- To check this data to get the center of the foot properly -->
  <!-- Inertia computation -->
  <!-- This macro has for purpose to define a left leg -->
  <!-- Include the front left leg -->
  <!-- Joint between the base and the ${prefix} upper leg -->
  <joint name="FL_HFE" type="revolute">
    <parent link="base_link"/>
    <child link="FL_upper_leg"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <!-- joints rotates around the y-axis -->
    <axis xyz="0 1 0"/>
    <!-- placement of the joint -->
    <origin rpy="0 0 0" xyz="0.19 0.1046 0"/>
    <!-- pybullet simulation parameters -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="FL_upper_leg">
    <!-- UPPER LEG LINK INERTIAL -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0793"/>
      <mass value="0.162"/>
      <inertia ixx="0.000469609" ixy="0.0" ixz="0.0" iyy="0.000469609" iyz="0.0" izz="1e-6"/>
    </inertial>
    <!-- UPPER LEG LINK VISUAL -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robot_properties_quadruped/meshes/obj/quadruped_upper_leg_left_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </visual>
    <!-- UPPER LEG LINK COLLISION -->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robot_properties_quadruped/meshes/obj/quadruped_upper_leg_left_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </collision>
  </link>
  <!-- END UPPER LEG LINK -->
  <!-- KFE: Joint between the upper leg and the lower leg -->
  <joint name="FL_KFE" type="revolute">
    <parent link="FL_upper_leg"/>
    <child link="FL_lower_leg"/>
    <parent link="base_link"/>
    <child link="FL_upper_leg"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <!-- joints rotates around the y-axis -->
    <axis xyz="0 1 0"/>
    <!-- placement of the joint -->
    <origin rpy="0 0 0" xyz="0 0.03745 -0.16"/>
    <!-- pybullet simulation parameters -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="FL_lower_leg">
    <!-- LOWER LEG LINK INERTIAL -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0756"/>
      <mass value="0.031"/>
      <inertia ixx="0.000087575" ixy="0.0" ixz="0.0" iyy="0.000087575" iyz="0.0" izz="1e-6"/>
    </inertial>
    <!-- LOWER LEG LINK VISUAL -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robot_properties_quadruped/meshes/obj/quadruped_lower_leg_left_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </visual>
    <!-- LOWER LEG LINK COLLISION -->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robot_properties_quadruped/meshes/obj/quadruped_lower_leg_left_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </collision>
  </link>
  <!-- END LOWER LEG LINK -->
  <!-- Include the front right leg -->
  <!-- Joint between the base and the ${prefix} upper leg -->
  <joint name="FR_HFE" type="revolute">
    <parent link="base_link"/>
    <child link="FR_upper_leg"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <!-- joints rotates around the y-axis -->
    <axis xyz="0 1 0"/>
    <!-- placement of the joint -->
    <origin rpy="0 0 0" xyz="0.19 -0.1046 0"/>
    <!-- pybullet simulation parameters -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="FR_upper_leg">
    <!-- UPPER LEG LINK INERTIAL -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0793"/>
      <mass value="0.162"/>
      <inertia ixx="0.000469609" ixy="0.0" ixz="0.0" iyy="0.000469609" iyz="0.0" izz="1e-6"/>
    </inertial>
    <!-- UPPER LEG LINK VISUAL -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robot_properties_quadruped/meshes/obj/quadruped_upper_leg_right_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </visual>
    <!-- UPPER LEG LINK COLLISION -->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robot_properties_quadruped/meshes/obj/quadruped_upper_leg_right_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </collision>
  </link>
  <!-- END UPPER LEG LINK -->
  <!-- KFE: Joint between the upper leg and the lower leg -->
  <joint name="FR_KFE" type="revolute">
    <parent link="FR_upper_leg"/>
    <child link="FR_lower_leg"/>
    <parent link="base_link"/>
    <child link="FR_upper_leg"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <!-- joints rotates around the y-axis -->
    <axis xyz="0 1 0"/>
    <!-- placement of the joint -->
    <origin rpy="0 0 0" xyz="0 -0.03745 -0.16"/>
    <!-- pybullet simulation parameters -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="FR_lower_leg">
    <!-- LOWER LEG LINK INERTIAL -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0756"/>
      <mass value="0.031"/>
      <inertia ixx="0.000087575" ixy="0.0" ixz="0.0" iyy="0.000087575" iyz="0.0" izz="1e-6"/>
    </inertial>
    <!-- LOWER LEG LINK VISUAL -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robot_properties_quadruped/meshes/obj/quadruped_lower_leg_right_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </visual>
    <!-- LOWER LEG LINK COLLISION -->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robot_properties_quadruped/meshes/obj/quadruped_lower_leg_right_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </collision>
  </link>
  <!-- END LOWER LEG LINK -->
  <!-- Include the hind left leg -->
  <!-- Joint between the base and the ${prefix} upper leg -->
  <joint name="HL_HFE" type="revolute">
    <parent link="base_link"/>
    <child link="HL_upper_leg"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <!-- joints rotates around the y-axis -->
    <axis xyz="0 1 0"/>
    <!-- placement of the joint -->
    <origin rpy="0 0 0" xyz="-0.19 0.1046 0"/>
    <!-- pybullet simulation parameters -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="HL_upper_leg">
    <!-- UPPER LEG LINK INERTIAL -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0793"/>
      <mass value="0.162"/>
      <inertia ixx="0.000469609" ixy="0.0" ixz="0.0" iyy="0.000469609" iyz="0.0" izz="1e-6"/>
    </inertial>
    <!-- UPPER LEG LINK VISUAL -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robot_properties_quadruped/meshes/obj/quadruped_upper_leg_left_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </visual>
    <!-- UPPER LEG LINK COLLISION -->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robot_properties_quadruped/meshes/obj/quadruped_upper_leg_left_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </collision>
  </link>
  <!-- END UPPER LEG LINK -->
  <!-- KFE: Joint between the upper leg and the lower leg -->
  <joint name="HL_KFE" type="revolute">
    <parent link="HL_upper_leg"/>
    <child link="HL_lower_leg"/>
    <parent link="base_link"/>
    <child link="HL_upper_leg"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <!-- joints rotates around the y-axis -->
    <axis xyz="0 1 0"/>
    <!-- placement of the joint -->
    <origin rpy="0 0 0" xyz="0 0.03745 -0.16"/>
    <!-- pybullet simulation parameters -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="HL_lower_leg">
    <!-- LOWER LEG LINK INERTIAL -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0756"/>
      <mass value="0.031"/>
      <inertia ixx="0.000087575" ixy="0.0" ixz="0.0" iyy="0.000087575" iyz="0.0" izz="1e-6"/>
    </inertial>
    <!-- LOWER LEG LINK VISUAL -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robot_properties_quadruped/meshes/obj/quadruped_lower_leg_left_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </visual>
    <!-- LOWER LEG LINK COLLISION -->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robot_properties_quadruped/meshes/obj/quadruped_lower_leg_left_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </collision>
  </link>
  <!-- END LOWER LEG LINK -->
  <!-- Include the hind right leg -->
  <!-- Joint between the base and the ${prefix} upper leg -->
  <joint name="HR_HFE" type="revolute">
    <parent link="base_link"/>
    <child link="HR_upper_leg"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <!-- joints rotates around the y-axis -->
    <axis xyz="0 1 0"/>
    <!-- placement of the joint -->
    <origin rpy="0 0 0" xyz="-0.19 -0.1046 0"/>
    <!-- pybullet simulation parameters -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="HR_upper_leg">
    <!-- UPPER LEG LINK INERTIAL -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0793"/>
      <mass value="0.162"/>
      <inertia ixx="0.000469609" ixy="0.0" ixz="0.0" iyy="0.000469609" iyz="0.0" izz="1e-6"/>
    </inertial>
    <!-- UPPER LEG LINK VISUAL -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robot_properties_quadruped/meshes/obj/quadruped_upper_leg_right_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </visual>
    <!-- UPPER LEG LINK COLLISION -->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robot_properties_quadruped/meshes/obj/quadruped_upper_leg_right_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </collision>
  </link>
  <!-- END UPPER LEG LINK -->
  <!-- KFE: Joint between the upper leg and the lower leg -->
  <joint name="HR_KFE" type="revolute">
    <parent link="HR_upper_leg"/>
    <child link="HR_lower_leg"/>
    <parent link="base_link"/>
    <child link="HR_upper_leg"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <!-- joints rotates around the y-axis -->
    <axis xyz="0 1 0"/>
    <!-- placement of the joint -->
    <origin rpy="0 0 0" xyz="0 -0.03745 -0.16"/>
    <!-- pybullet simulation parameters -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="HR_lower_leg">
    <!-- LOWER LEG LINK INERTIAL -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0756"/>
      <mass value="0.031"/>
      <inertia ixx="0.000087575" ixy="0.0" ixz="0.0" iyy="0.000087575" iyz="0.0" izz="1e-6"/>
    </inertial>
    <!-- LOWER LEG LINK VISUAL -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robot_properties_quadruped/meshes/obj/quadruped_lower_leg_right_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </visual>
    <!-- LOWER LEG LINK COLLISION -->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robot_properties_quadruped/meshes/obj/quadruped_lower_leg_right_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </collision>
  </link>
  <!-- END LOWER LEG LINK -->
</robot>
